nixos/modules/services/web-servers/keter/config.nix

```diff
@@ -0,0 +1,74 @@
+{ globalketerConfig, keterRoot, writeShellScriptBin,
+  writeTextFile, lib, bundleCfg, callPackage, coreutils
+  , keter, ...
+}:
+
+let
+  incoming = "${keterRoot}/incoming";
+
+
+  globalKeterConfigFile = writeTextFile {
+    name = "keter-config.yml";
+    text = (lib.generators.toYAML { } globalketerConfig);
+  };
+
+  # If things are expected to change often, put it in the bundle!
+  bundle = callPackage ./bundle.nix
+    (bundleCfg // { exePkg = executable; });
+
+  executable = writeShellScriptBin "${bundleCfg.executable}" ''
+    #!/usr/bin/env bash
+    export PATH=$PATH:${lib.makeBinPath [ coreutils ]}
+    set -e
+    ${bundleCfg.secretScript}
+    set -xe
+    ${bundleCfg.publicScript}
+    exec ${bundleCfg.package}/bin/${bundleCfg.executable}
+  '';
+
+in {
+  # This is keter, which loads our app on deploy, and checks if it is healthy
+  # be very careful putting anything in here that changes,
+  # any change to the systemd unit can cause brief windows of downtime
+  # because nixos has to restart the unit.
```

Comment:
Why is this here? Users might not look at this file before using it.

Reply:
Where should I put this?